swift/ios-shoppe-demo/Resources/Constants.swift

```diff
@@ -0,0 +1,83 @@
+        static let productSummaryCell = "ProductSummaryCell"
+        static let productSummaryTopCell = "ProductSummaryTopCell"
+        static let productSummaryUserDetailCell = "ProductSummaryUserDetailCell"
+        static let productSummaryTotalCost = "ProductSummaryTotalCost"
+        static let loginButtonCell = "LoginButtonViewCell"
+    }
+
+    struct CellID {
+        static let productView = "cell"
+        static let checkout = "checkout"
+        static let quantity = "quantity"
+        static let title = "title"
+        static let checkoutBox = "checkoutBox"
+        static let textField = "textField"
+        static let thankYou = "thankYou"
+        static let userDetails = "userDetails"
+        static let totalCell = "totalCell"
+        static let login = "login"
+    }
+
+    struct Image {
+        static let cartIcon = "shopping_cart"
+    }
+}
+
+typealias Registry = ([String], [String])
+
+let cartTableViewRegistry: Registry = (nibArr: [nib.cartTableViewCell,
+                                                            nib.cartQuant,
+                                                            nib.largeLabel],
+                                       cellIDArr: [cellID.checkout,
+                                                               cellID.quantity,
+                                                               cellID.title])
+
+let checkoutTableViewRegistry: Registry = (nibArr: [nib.checkoutTableView,
+                                                                nib.largeLabel,
+                                                                nib.cartTableViewCell,
+                                                                nib.checkBoxButton],
+                                           cellIDArr: [cellID.textField,
+                                                                   cellID.title,
+                                                                   cellID.checkout,
+                                                                   cellID.checkoutBox])
+
+let productSummaryRegistry: Registry = (nibArr: [nib.productSummaryCell,
+                                                     nib.productSummaryTopCell,
+                                                     nib.productSummaryUserDetailCell,
+                                                     nib.productSummaryTotalCost],
+                                        cellIDArr: [cellID.quantity,
+                                                        cellID.thankYou,
+                                                        cellID.userDetails,
+                                                        cellID.totalCell])
+let loginViewControllerRegistry: Registry = (nibArr: [nib.loginButtonCell,
+                                             nib.checkoutTableView],
+                                             cellID: [cellID.login,
+                                                      cellID.textField])
```

Comment:
When reaching for labeled tuples that all have the same name, it's a code smell indicating that this should be its own dedicated type. Consider changing it to a `struct` like:

```swift
struct Registry {
    var nib: [String]
    var cellID: [String]

    static let cartTableView = Registry(
        nib: [nib.cartTableViewCell, nib.cartQuant, nib.largeLabel],
        cellID: [cellID.checkout, cellID.quantity, cellID.title]
    )

    static let checkoutTableView = Registry(
        nib: [nib.checkoutTableView, nib.largeLabel, nib.cartTableViewCell, nib.checkBoxButton],
        cellID: [cellID.textField, cellID.title, cellID.checkout, cellID.checkoutBox]
    )

    static let productSummary = Registry(
        nib: [nib.productSummaryCell, nib.productSummaryTopCell, nib.productSummaryUserDetailCell, nib.productSummaryTotalCost],
        cellID: [cellID.quantity, cellID.thankYou, cellID.userDetails, cellID.totalCell]
    )

    static let loginViewController = Registry(
        nib: [nib.loginButtonCell, nib.checkoutTableView],
        cellID: [cellID.login, cellID.textField]
    )
}
```

This should probably go in its own file.

Also, it's usually extra noise to put the name of the type in a property's name.